File: src/main/java/co/zpdev/bots/simpletickets/commands/Close.java
package co.zpdev.bots.simpletickets.commands;

import co.zpdev.bots.core.command.Command;
import co.zpdev.bots.simpletickets.SimpleTickets;
import net.dv8tion.jda.core.EmbedBuilder;
import net.dv8tion.jda.core.entities.Message;
import net.dv8tion.jda.core.entities.MessageEmbed;
import net.dv8tion.jda.core.entities.TextChannel;

import java.time.Instant;

public class Close {

    @Command(aliases = "close")
    public void onCommand(Message message) {
        if (!message.getMember().isOwner() && message.getMember().getRoles().stream().noneMatch(r -> r.getId().equals("388055883172806656"))) return;
        if (!message.getChannel().getName().startsWith("ticket-")) return;

        message.delete().queue();

        TextChannel c = message.getTextChannel();
        c.delete().complete();

        MessageEmbed log = new EmbedBuilder().setColor(SimpleTickets.embed)
                .setAuthor("Ticket Closed")
                .setDescription("#" + c.getName() + " closed by **" + message.getMember().getEffectiveName() + "**.")
                .setTimestamp(Instant.now()).build();
        message.getGuild().getTextChannelById(SimpleTickets.logs).sendMessage(log).queue();
    }

}

File: src/main/kotlin/com/zp4rker/simpletickets/commands/Close.kt
package com.zp4rker.simpletickets.commands

import com.zp4rker.core.discord.command.Command
import com.zp4rker.simpletickets.SimpleTickets
import com.zp4rker.simpletickets.embedColour
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.entities.Guild
import net.dv8tion.jda.api.entities.Message
import net.dv8tion.jda.api.entities.TextChannel
import java.time.Instant

object Close : Command(aliases = arrayOf("close"), description = "Closes an open ticket.", autoDelete = true) {
    override fun handle(message: Message, channel: TextChannel, guild: Guild, args: List<String>) {
        val member = message.member ?: return
        val logs = guild.getTextChannelById(SimpleTickets.logs) ?: return

        if (member.roles.none { SimpleTickets.roles.contains(it.idLong) }) return
        if (!channel.name.startsWith("ticket-")) return

        channel.delete().queue {
            EmbedBuilder().setColor(embedColour).apply {
                setTitle("Ticket closed")
                setDescription("#${channel.name} closed by **${member.user.asTag}**.")
                setTimestamp(Instant.now())
            }.build().apply { logs.sendMessage(this).queue() }
        }
    }
}